gradle/config/pmd/pmd.xml

```diff
@@ -0,0 +1,202 @@
+        <!--
+          Terra: this rule is overly sensitive on method naming.
+        -->
+        <exclude name="LinguisticNaming"/>
+        <exclude name="LocalHomeNamingConvention"/> <!-- earlier j2ee group-->
+        <exclude name="LocalInterfaceSessionNamingConvention"/> <!-- earlier j2ee group-->
+        <exclude name="LocalVariableCouldBeFinal"/>
+        <exclude name="LongVariable"/>
+        <exclude name="MDBAndSessionBeanNamingConvention"/> <!-- earlier j2ee group-->
+        <!--
+          Terra: final on method arguments is not required.
+        -->
+        <exclude name="MethodArgumentCouldBeFinal"/>
+        <exclude name="OnlyOneReturn"/> <!-- disabled in favour of checkstyle ReturnCount check -->
+        <!--
+          Terra: overly sensitive rule with marginal value.
+        -->
+        <exclude name="PrematureDeclaration"/>
+        <exclude name="RemoteInterfaceNamingConvention"/> <!-- earlier j2ee group-->
+        <exclude name="RemoteSessionInterfaceNamingConvention"/> <!-- earlier j2ee group-->
+        <exclude name="ShortClassName"/>
+        <exclude name="ShortMethodName"/>
+        <exclude name="ShortVariable"/>
+        <exclude name="TooManyStaticImports"/>
+        <exclude name="UselessParentheses"/>
+    </rule>
+    <rule ref="category/java/codestyle.xml/FieldNamingConventions">
+        <properties>
+            <property name="enumConstantPattern" value="[A-Z][a-zA-Z_0-9]*"/>
+        </properties>
+    </rule>
+
+
+    <!-- Design (https://pmd.github.io/pmd-6.0.1/pmd_rules_java_design.html) -->
+    <rule ref="category/java/design.xml">
+        <exclude name="AvoidCatchingGenericException"/>
+        <exclude name="AvoidUncheckedExceptionsInSignatures"/>
+        <!--
+          Terra: complexity measures are squishy and vague. Prefer to catch complexity in code review.
+        -->
+        <exclude name="CyclomaticComplexity"/>
+        <exclude name="DataClass"/>
+        <exclude name="ExcessiveClassLength"/> <!-- replaced by NcssCount -->
+        <exclude name="ExcessiveMethodLength"/> <!-- replaced by NcssCount -->
+        <exclude name="LawOfDemeter"/>
+        <exclude name="LoosePackageCoupling"/>
+        <exclude name="ModifiedCyclomaticComplexity"/>
+        <!--
+          Terra: complexity measures are squishy and vague. Prefer to catch complexity in code review.
+        -->
+        <exclude name="NcssCount"/>
+        <exclude name="SignatureDeclareThrowsException"/>
+        <exclude name="StdCyclomaticComplexity"/>
+    </rule>
+    <rule ref="category/java/design.xml/AvoidDeeplyNestedIfStmts">
+        <properties>
+            <property name="problemDepth" value="4"/>
+        </properties>
+    </rule>
+    <!--
+      Terra: complexity measures are squishy and vague. Prefer to catch complexity in code review.
+
+      (The commented-out sections below were present in the default geneated XML. They are kept
+      here in commented form for clarity and provenance.)

```

Comment:
I don't get why they're commented out instead of deleted. Are we ever going to refer to the default generated XML again?

Reply:
Yeah, I understand the aversion to checking in commented-out stuff. But for this type of config, where we're taking something "stock" and making targeted changes particular to our environment, I think deletions are just as relevant as additions.

Imagine someone a year from now wants to upgrade to a new major version of the quality plugin. Keeping this here makes it a bit easier to understand what opinions we had re: specific suppressions and why, and re-apply the same changes if needed to a newly-generated stock config.